### PR TITLE
fix: use correct link for reporting false positives in blockaid banner

### PR DESCRIPTION
**Pull request by jpuri**
## **Description**
Fix the link used for reporting false positives on blockaid banner and also pass more data to it.

## **Related issues**

Fixes: MetaMask/MetaMask-planning#1697

## **Manual testing steps**

1. Enables blockaid in mobile app
2. Go to test dapp and submit malicious transaction
3. Click "Report the issue" link. It should open in right site with data prefilled.

## **Screenshots/Recordings**

https://github.com/MetaMask/metamask-mobile/assets/2182307/98d32b27-4617-41bc-b26c-63568bbe4d7c

## **Pre-merge author checklist**

- [X] I’ve followed [MetaMask Coding Standards](https://github.com/MetaMask/metamask-mobile/blob/main/.github/guidelines/CODING_GUIDELINES.md).
- [X] I've clearly explained what problem this PR is solving and how it is solved.
- [X] I've linked related issues
- [X] I've included manual testing steps
- [X] I've included screenshots/recordings if applicable
- [X] I’ve included tests if applicable
- [ ] I’ve documented my code using [JSDoc](https://jsdoc.app/) format if applicable
- [X] I’ve applied the right labels on the PR (see [labeling guidelines](https://github.com/MetaMask/metamask-mobile/blob/main/.github/guidelines/LABELING_GUIDELINES.md)). Not required for external contributors.
- [X] I’ve properly set the pull request status:
  - [X] In case it's not yet "ready for review", I've set it to "draft".
  - [ ] In case it's "ready for review", I've changed it from "draft" to "non-draft".

## **Pre-merge reviewer checklist**

- [ ] I've manually tested the PR (e.g. pull and build branch, run the app, test code being changed).
- [ ] I confirm that this PR addresses all acceptance criteria described in the ticket it closes and includes the necessary testing evidence such as recordings and or screenshots.

### CODE REVIEW

**Comment by jpuri** [comment]
@SocketSecurity ignore react-native-gzip@1.1.0
@SocketSecurity ignore deprecated-react-native-prop-types@3.0.2

File: ppom/package.json [hunk 1]
```diff
@@ -20,7 +20,8 @@
     "lint": "eslint '**/*.{js,ts,tsx}'",
     "build-html": "webpack --mode production --config ./webpack.config.html.js",
     "build-html-js": "webpack --mode production --config ./webpack.config.html-js.js",
-    "build": "yarn clean && yarn build-html && yarn build-html-js",
+    "build-version": "webpack --mode production --config ./webpack.config.version.js",
+    "build": "yarn clean && yarn build-html && yarn build-html-js && yar build-version",
```
Comment (blackdevelopa) [inline ppom/package.json:24]:
```suggestion
    "build": "yarn clean && yarn build-html && yarn build-html-js && yarn build-version",
```

File: app/constants/network.js [hunk 2]
```diff
@@ -32,3 +32,12 @@ export const NETWORKS_CHAIN_ID = {
   LINEA_MAINNET: '59144',
   ZKSYNC_ERA: '324',
 };
+
+export const NETWORK_NAMES = {
```
Comment (segun) [inline app/constants/network.js:36]: I'm guessing these are the networks supported by blockaid, if yes can we move them to a `shared/constants/security-provider.ts` and maybe change the name to say `SUPPORTED_NETWORK_NAMES`

If not, then what happens when we do this
`chain: NETWORK_NAMES[chainId],` for a chainId not in this list, does the URL still work when chain is undefined?

Comment (jpuri) [reply app/constants/network.js]: Yes this is set of supported networks. I am unable to fix `shared/constants/security-provider.ts` in mobile repo but I renamed the constant and moved it to BlockaidBanner component.

File: package.json [hunk 3]
```diff
@@ -310,6 +310,7 @@
     "react-native-fs": "^2.16.6",
     "react-native-gesture-handler": "^1.10.3",
     "react-native-get-random-values": "^1.8.0",
+    "react-native-gzip": "^1.1.0",
```
Comment (digiwand) [inline package.json:313]: Q: do we have alternative solutions to adding this new [npm package](https://www.npmjs.com/package/react-native-gzip?activeTab=readme)?

**Review by segun — DISMISSED** [review]

File: package.json (hunk 3 above)
Comment (jpuri) [reply package.json:313]: No, I did searched but could not find any other solution

File: app/components/UI/BlockaidBanner/BlockaidBanner.tsx [hunk 4]
```diff
@@ -35,11 +36,26 @@ import {
...
-  BLOCKAID_SUPPORT_LINK,
+  FALSE_POSITIVE_REPORT_BASE_URL,
+  UTM_SOURCE,
 } from '../../../constants/urls';
 import { isMainnetByChainId } from '../../../util/networks';
-import { selectChainId } from '../../../selectors/networkController';
 import { selectIsSecurityAlertsEnabled } from '../../../selectors/preferencesController';
+import BlockaidVersionInfo from '../../../lib/ppom/blockaid-version';
+
+const BLOCKAID_SUPPORTED_NETWORK_NAMES = {
```
Comment (blackdevelopa) [inline app/components/UI/BlockaidBanner/BlockaidBanner.tsx:46]: Can we move this to `../util/network`?

**Review by blackdevelopa — APPROVED** [review]
Small nit. LGTM

**Review by segun — APPROVED** [review]

File: app/components/UI/BlockaidBanner/BlockaidBanner.tsx (hunk 4 above)
Comment (jpuri) [reply app/components/UI/BlockaidBanner/BlockaidBanner.tsx:46]: I will fix it in a following PR.

**Comment by jpuri** [comment]
@SocketSecurity ignore npm/deprecated-react-native-prop-types@3.0.2

**Comment by jpuri** [comment]
https://app.bitrise.io/app/be69d4368ee7e86d/pipelines/c8c06d3e-4ee3-49b4-860b-e21171c43d93

**Pull request merged**